ee/management/commands/backfill_has_full_snapshot.py

```diff
@@ -0,0 +1,34 @@
+from django.core.management.base import BaseCommand
+
+from ee.clickhouse.client import sync_execute
+from ee.clickhouse.sql.session_recording_events import SESSION_RECORDING_EVENTS_TABLE
+from posthog.settings import CLICKHOUSE_CLUSTER, CLICKHOUSE_REPLICATION
+
+
+class Command(BaseCommand):
+    help = "Backfill the has_full_snapshot materialized column on session_recording_events"
+
+    def handle(self, *args, **options):
+        updated_table = (
+            f"sharded_{SESSION_RECORDING_EVENTS_TABLE}" if CLICKHOUSE_REPLICATION else SESSION_RECORDING_EVENTS_TABLE
+        )
+
+        # Hack from https://github.com/ClickHouse/ClickHouse/issues/19785
+        # Note that for this to work all inserts should list columns explicitly
+        # Improve this if https://github.com/ClickHouse/ClickHouse/issues/27730 ever gets resolved
+        sync_execute(
+            f"""
+            ALTER TABLE {updated_table}
+            ON CLUSTER {CLICKHOUSE_CLUSTER}
+            MODIFY COLUMN has_full_snapshot
+            BOOLEAN DEFAULT JSONExtractBool(snapshot_data, 'has_full_snapshot')
+            """
+        )
+
+        sync_execute(
+            f"""
+            ALTER TABLE {updated_table}
+            ON CLUSTER {CLICKHOUSE_CLUSTER}
+            UPDATE has_full_snapshot = JSONExtractBool(snapshot_data, 'has_full_snapshot') WHERE 1=1
+            """
+        )
```

Comment:
This is following how we backfill other materialized columns, but I really don't understand why/how this works. Also, do we need to worry about this command potentially leaving customers in different states? Meaning those who have run this command could see different behavior from those who haven't.